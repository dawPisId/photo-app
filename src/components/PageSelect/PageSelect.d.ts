export interface PageSelectProps {
  currentPage: number;
  currentPageSetter: (value: number) => void;
  pagesCount: number;
  itemCount: number;
  itemsPerPageSetter: (value: number) => void;
}
