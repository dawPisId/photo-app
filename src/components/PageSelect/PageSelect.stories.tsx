import { ComponentMeta, ComponentStory } from "@storybook/react";

import { BrowserRouter } from "react-router-dom";
import PageSelect from "./PageSelect";
import React from "react";

export default {
  title: "Photo App/PageSelect",
  component: PageSelect,
} as ComponentMeta<typeof PageSelect>;

const Template: ComponentStory<typeof PageSelect> = (args) => (
  <BrowserRouter>
    <PageSelect {...args} />
  </BrowserRouter>
);

export const Main = Template.bind({});
Main.args = {
  currentPage: 10,
  currentPageSetter: () => null,
  pagesCount: 50,
  itemCount: 20,
  itemsPerPageSetter: () => null,
};
