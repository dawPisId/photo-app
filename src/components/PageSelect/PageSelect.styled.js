import styled from "styled-components";

export const SelectContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1rem;
  svg {
    &.MuiSvgIcon-root {
      font-size: 1rem;
      cursor: pointer;
    }
  }
`;

export const SelectItem = styled.div`
  margin: 0.5rem;
  padding: 0.5rem 0.9rem;
  cursor: pointer;
  border-radius: 500px;
  &.active {
    background-color: lightgrey;
  }
`;

export const ItemCountWrapper = styled.div`
  position: relative;
`;

export const ItemCountButton = styled.div`
  display: inline-flex;
  align-self: center;
  align-items: center;
  position: relative;
  border: 1px solid lightgrey;
  border-radius: 5px;
  padding-right: 0.5rem;
  margin-left: 2rem;
  cursor: pointer;
  &:active {
    background-color: #d9d9d9;
  }
`;

export const ItemCountList = styled.div`
  font-size: 0.76rem;
  margin: 0rem;
  position: absolute;
  border: 1px solid lightgrey;
  border-top: none;
  border-radius: 5px;
  right: 0px;
  flex-direction: column;
  &.active {
    display: flex;
  }
  &.hidden {
    display: none;
  }
`;

export const ItemCountListItem = styled.div`
  padding: 0.5rem 1.84rem 0.5rem 1.84rem;
  cursor: pointer;
  &:hover {
    background-color: #d9d9d9;
  }
`;

export const ItemCountNumber = styled.div`
  display: inline-block;
  padding: 0.5rem 0.5rem 0.5rem 1.5rem;
  background-color: rgba(0, 0, 0, 0);
  font-size: 1rem;
  cursor: pointer;
  font-family: "Roboto", sans-serif;
  border: none;
  border-radius: 2px;
`;

export const ItemCountLabel = styled.span`
  position: absolute;
  color: grey;
  top: -0.65rem;
  left: 1rem;
  font-size: 0.75rem;
`;
