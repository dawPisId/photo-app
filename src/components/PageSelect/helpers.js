export const zoom = (setZoom, setUrl, imgUrl, setLabel, label) => {
  setUrl(imgUrl);
  setLabel(label);
  setZoom(true);
};

export const range = (start, end) => {
  let length = end - start + 1;
  return Array.from({ length }, (_, idx) => idx + start);
};

export const prev = (currentPage, setCurrentPage) => {
  if (currentPage > 1) {
    setCurrentPage(currentPage - 1);
  }
};

export const next = (currentPage, setCurrentPage, pageCount) => {
  if (currentPage < pageCount) {
    setCurrentPage(currentPage + 1);
  }
};

export const changeItemCount = (itemCount, setItemCount, setDropdownToggle) => {
  setItemCount(itemCount);
  setDropdownToggle(false);
};

export const counts = [10, 20, 50];
