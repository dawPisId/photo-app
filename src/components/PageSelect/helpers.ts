export const zoom = (
  setZoom: (bool: boolean) => void,
  setUrl: (value: string) => void,
  imgUrl: string,
  setLabel: (value: string) => void,
  label: string
) => {
  setUrl(imgUrl);
  setLabel(label);
  setZoom(true);
};

export const range = (start: number, end: number) => {
  let length = end - start + 1;
  return Array.from({ length }, (_, idx) => idx + start);
};

export const prev = (
  currentPage: number,
  setCurrentPage: (number: number) => void
) => {
  if (currentPage > 1) {
    setCurrentPage(currentPage - 1);
  }
};

export const next = (
  currentPage: number,
  setCurrentPage: (number: number) => void,
  pageCount: number
) => {
  if (currentPage < pageCount) {
    setCurrentPage(currentPage + 1);
  }
};

export const changeItemCount = (
  itemCount: number,
  setItemCount: (count: number) => void,
  setDropdownToggle: (bool: boolean) => void
) => {
  setItemCount(itemCount);
  setDropdownToggle(false);
};

export const counts = [12, 16, 20];
