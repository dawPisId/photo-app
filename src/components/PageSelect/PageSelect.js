import {
  ItemCountButton,
  ItemCountLabel,
  ItemCountList,
  ItemCountListItem,
  ItemCountNumber,
  ItemCountWrapper,
  SelectContainer,
} from "./PageSelect.styled";
import React, { useState } from "react";
import { changeItemCount, counts } from "./helpers";

import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import PageSelectItems from "../PageSelectItems/PageSelectItems";

function PageSelect({
  currentPage,
  currentPageSetter,
  pagesCount,
  itemCount,
  itemsPerPageSetter,
}) {
  const [dropdownToggle, setDropdownToggle] = useState(false);
  return (
    <div>
      <SelectContainer>
        <PageSelectItems
          currentPage={currentPage}
          setCurrentPage={currentPageSetter}
          pageCount={pagesCount}
        />
        <ItemCountWrapper>
          <ItemCountButton onClick={() => setDropdownToggle(!dropdownToggle)}>
            <ItemCountLabel>Per page</ItemCountLabel>
            <ItemCountNumber>{itemCount}</ItemCountNumber>
            <ArrowDropDownIcon />
          </ItemCountButton>
          <ItemCountList className={dropdownToggle ? "active" : "hidden"}>
            {counts.map((number, index) => (
              <ItemCountListItem
                key={index}
                onClick={() =>
                  changeItemCount(number, itemsPerPageSetter, setDropdownToggle)
                }
              >
                {number}
              </ItemCountListItem>
            ))}
          </ItemCountList>
        </ItemCountWrapper>
      </SelectContainer>
    </div>
  );
}

export default PageSelect;
