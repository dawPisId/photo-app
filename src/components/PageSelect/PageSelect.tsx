import {
  ItemCountButton,
  ItemCountLabel,
  ItemCountList,
  ItemCountListItem,
  ItemCountNumber,
  ItemCountWrapper,
  SelectContainer,
} from "./PageSelect.styled";
import React, { useState } from "react";
import { changeItemCount, counts } from "./helpers";

import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import PageSelectItems from "../PageSelectItems/PageSelectItems";
import { PageSelectProps } from './PageSelect.d';

const PageSelect: React.FC<PageSelectProps> = ({
  currentPage,
  currentPageSetter,
  pagesCount,
  itemCount,
  itemsPerPageSetter,
}) => {
  const [dropdownToggle, setDropdownToggle] = useState(false);
  return (
    <div>
      <SelectContainer>
        <PageSelectItems
          currentPage={currentPage}
          setCurrentPage={currentPageSetter}
          pageCount={pagesCount}
        />
        <ItemCountWrapper>
          <ItemCountButton data-testid = "testButton" onClick={() => setDropdownToggle(!dropdownToggle)}>
            <ItemCountLabel>Per page</ItemCountLabel>
            <ItemCountNumber>{itemCount}</ItemCountNumber>
            <ArrowDropDownIcon />
          </ItemCountButton>
          <ItemCountList data-testid = "testList" className={dropdownToggle ? "active" : "hidden"}>
            {counts.map((number, index) => (
              <ItemCountListItem
                key={index}
                onClick={() =>
                  changeItemCount(number, itemsPerPageSetter, setDropdownToggle)
                }
              >
                {number}
              </ItemCountListItem>
            ))}
          </ItemCountList>
        </ItemCountWrapper>
      </SelectContainer>
    </div>
  );
};

export default PageSelect;
