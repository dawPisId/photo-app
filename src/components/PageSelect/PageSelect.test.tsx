import PageSelect from "./PageSelect";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <PageSelect
        currentPage={10}
        currentPageSetter={() => null}
        pagesCount={50}
        itemCount={20}
        itemsPerPageSetter={() => null}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
