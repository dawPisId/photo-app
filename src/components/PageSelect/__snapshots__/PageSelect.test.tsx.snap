// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-bdnxRM ibzVqB"
  >
    <div
      className="sc-fujyAs blTUGX"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <div
        className="sc-pNWdM gDosGb"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="sc-pNWdM gDosGb"
      >
        ...
      </div>
      <div
        className="sc-pNWdM gDosGb"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="sc-pNWdM gDosGb active"
        onClick={[Function]}
      >
        10
      </div>
      <div
        className="sc-pNWdM gDosGb"
        onClick={[Function]}
      >
        11
      </div>
      <div
        className="sc-pNWdM gDosGb"
      >
        ...
      </div>
      <div
        className="sc-pNWdM gDosGb"
        onClick={[Function]}
      >
        50
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
    </div>
    <div
      className="sc-dlnjwi fHmCXS"
    >
      <div
        className="sc-hKFxyN evguJw"
        data-testid="testButton"
        onClick={[Function]}
      >
        <span
          className="sc-iCoGMd GYIbd"
        >
          Per page
        </span>
        <div
          className="sc-gKAaRy kuezKY"
        >
          20
        </div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <div
        className="sc-eCApnc bbMYLx hidden"
        data-testid="testList"
      >
        <div
          className="sc-jSFjdj bXtTsJ"
          data-testid="testListItem"
          onClick={[Function]}
        >
          12
        </div>
        <div
          className="sc-jSFjdj bXtTsJ"
          data-testid="testListItem"
          onClick={[Function]}
        >
          16
        </div>
        <div
          className="sc-jSFjdj bXtTsJ"
          data-testid="testListItem"
          onClick={[Function]}
        >
          20
        </div>
      </div>
    </div>
  </div>
</div>
`;
